fidelius-sdk/src/main/java/org/finra/fidelius/FideliusClient.java

```diff
@@ -17,114 +17,128 @@
         }
 
-        AmazonDynamoDBClientBuilder ddbBuilder = AmazonDynamoDBClientBuilder.standard()
-                .withCredentials(provider)
-                .withClientConfiguration(clientConf);
+        DynamoDbClientBuilder dynamoDbBuilder = DynamoDbClient.builder()
+                .credentialsProvider(provider)
+                .overrideConfiguration(clientConf);
 
-        AWSKMSClientBuilder kmsBuilder = AWSKMSClientBuilder.standard()
-                .withCredentials(provider)
-                .withClientConfiguration(kmsEc2ClientConfiguration);
+        KmsClientBuilder kmsBuilder = KmsClient.builder()
+                .credentialsProvider(provider)
+                .overrideConfiguration(kmsEc2ClientConfiguration);
 
-        AWSSecurityTokenServiceClientBuilder stsBuilder =  AWSSecurityTokenServiceClientBuilder.standard()
-                .withClientConfiguration(clientConf)
-                .withCredentials(provider);
+        StsClientBuilder stsBuilder =  StsClient.builder()
+                .overrideConfiguration(clientConf)
+                .credentialsProvider(provider);
 
-        AmazonEC2ClientBuilder clientBuilder = AmazonEC2ClientBuilder.standard()
-                .withCredentials(provider)
-                .withClientConfiguration(kmsEc2ClientConfiguration);
+        Ec2ClientBuilder clientBuilder = Ec2Client.builder()
+                .credentialsProvider(provider)
+                .overrideConfiguration(kmsEc2ClientConfiguration);
 
-        AWSLambdaClientBuilder lambdaClientBuilder = AWSLambdaClientBuilder.standard()
-                .withClientConfiguration(clientConf)
-                .withCredentials(provider);
+        LambdaClientBuilder lambdaClientBuilder = LambdaClient.builder()
+                .credentialsProvider(provider)
+                .overrideConfiguration(clientConf);
 
         if(region != null){
-            Regions regionEnum = Regions.fromName(region);
-            ddbBuilder.withRegion(regionEnum);
-            kmsBuilder.withRegion(regionEnum);
-            stsBuilder.withRegion(regionEnum);
-            clientBuilder.withRegion(regionEnum);
-            lambdaClientBuilder.withRegion(regionEnum);
+            software.amazon.awssdk.regions.Region awsRegion = software.amazon.awssdk.regions.Region.of(region);
+            dynamoDbBuilder = dynamoDbBuilder.region(awsRegion);
+            kmsBuilder = kmsBuilder.region(awsRegion);
+            stsBuilder = stsBuilder.region(awsRegion);
+            clientBuilder = clientBuilder.region(awsRegion);
+            lambdaClientBuilder = lambdaClientBuilder.region(awsRegion);
         }
-        lambda = lambdaClientBuilder.build();
-        client = clientBuilder.build();
-        awsSecurityTokenService = stsBuilder.build();
-        jCredStash = new JCredStash(ddbBuilder.build(), kmsBuilder.build(), awsSecurityTokenService);
+        if(envConfig.hasProxyEnv()) {
```

Comment:
Not sure if I understand this if condition, don't we have to initialize all the clients if proxy env is provided and if not provided, like if.....else....

Reply:
We're still building the Clients on lines 128-131. For AWS SDK v2, client builders like the DynamoDbClientBuilder need an additional HTTP Client built-in for proxy configuration to work. The if-block there takes care of that if proxy configuration is provided but skips if it's not.